modules/indexing/src/test/java/org/apache/ignite/cache/query/IndexQueryPaginationTest.java

```diff
@@ -0,0 +1,150 @@
+package org.apache.ignite.cache.query;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.cache.Cache;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.IgniteDataStreamer;
+import org.apache.ignite.cache.CacheMode;
+import org.apache.ignite.cache.query.annotations.QuerySqlField;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.TestRecordingCommunicationSpi;
+import org.apache.ignite.internal.processors.cache.query.GridCacheQueryRequest;
+import org.apache.ignite.internal.processors.cache.query.GridCacheQueryResponse;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static org.apache.ignite.cache.query.ThinClientIndexQueryTest.getFilteredMessages;
+
+/** */
+public class IndexQueryPaginationTest extends GridCommonAbstractTest {
+    /** */
+    private static final int PAGE_SIZE = 512;
+
+    /** */
+    private static final int NODES = 2;
+
+    /** */
+    private Map<Integer, Integer> entriesAndReqs;
+
+    /** */
+    private Ignite grid;
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        entriesAndReqs = new HashMap<Integer, Integer>() {{
```

Comment:
Let's make test parameterized

Reply:
The test has been added to `IndexQueryTestSuite` and rewritten as parameterized.